src/particles/ImpactXParticleContainer.cpp

```diff
@@ -121,4 +125,78 @@ namespace impactx
         return {xyz_min[0], xyz_min[1], xyz_min[2], xyz_max[0], xyz_max[1], xyz_max[2]};
     }
 
+    void
+    DepositCharge (ImpactXParticleContainer & pc, amrex::MultiFab* rho, int icomp) // ImpactXParticleContainer & pc)
+    {
+        // loop over refinement levels
+        int const nLevel = 1;
+        for (int lev = 0; lev < nLevel; ++lev) {
+            // get simulation geometry information
+            //const amrex::Geometry& gm = this->Geom(lev);
+            //const auto prob_lo = gm.ProbLo();
+
```

Comment:
If we'd like to support OpenMP threading for CPU execution, then the #pragma omp parallel region should go here, including the local_rho_fab as a thread private variable.

Reply:
added :)